pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -711,3 +710,17 @@ func (c *Cluster) resolveNode(nodeName string) *rookalpha.Node {
 
 	return rookNode
 }
+
+func (c *Cluster) getOSDPropsForPVC(pvcName string) (osdProperties, error) {
+	for _, volumeSource := range c.ValidStorage.VolumeSources {
```

Comment:
Something to keep in mind is that if there are many OSDs on PVCs, we could have an issue with the loop through all PVCs for every OSD (n^2 algorithm). Is there not a more direct lookup for the pvc source? If not, we may need to build create such a map.

Reply:
yeah agreed. But currently, I don't think we have any other way for lookup.

Reply:
Ok, i'll go ahead and merge and open an issue to follow up.